pennylane_lightning_gpu/src/util/DataBuffer.hpp

```diff
@@ -37,6 +37,7 @@ template <class GPUDataT, class DevTagT = int> class DataBuffer {
             PL_CUDA_IS_SUCCESS(
                 cudaMalloc(reinterpret_cast<void **>(&gpu_buffer_),
                            sizeof(GPUDataT) * length));
+            zeroInit();
```

Comment:
I don't think we want to always set the buffer to 0. Sometimes we just need the memory, and want to copy the statevector without setting it, as is done in several places in the adjoint pipeline.

Can we do this another way?

Reply:
Done. Just remove all the `initZeros()` in the ctors. The `initZeros()` is only called when `setBasisState` and  `setStateVector`.